### ISSUE TITLE
TinyJson does not properly serialize long data type

### ISSUE THREAD

**Issue by thegoldenmule** [heroiclabs/nakama-unity#90]
### Repro steps:

The Nakama TinyJson library does not properly support the `long` data type.

```
public class Foo
{
    public long Bar;
}

var a = new Foo { Bar = 100 };
a.ToJson(); // { "Bar": {} }
```

We have switched to using Newtonsoft JSON which properly serializes/deserializes longs.

**Comment by novabyte** [comment]
Hmmm this is an odd one. We use the TinyJson library from [here](https://github.com/zanders3/json). We'll have a look to see what we can do to sort it.

**Comment by thegoldenmule** [comment]
Looks like it actually doesn't support _most_ primitives (unsigned and signed, short, long, etc). I can create a PR in that repo if it would be helpful.

**Comment by novabyte** [comment]
@thegoldenmule That would be awesome if you have time 🙏

**Comment by thegoldenmule** [comment]
@novabyte https://github.com/zanders3/json/pull/36

**Comment by thegoldenmule** [comment]
@novabyte The json PR has been merged to master btw.

**Comment by novabyte** [comment]
Ah yes - good reminder. I'll re-import that code into the .NET client so we can cut a release of the Unity client.

**Comment by novabyte** [comment]
I've updated the TinyJson library in the .NET client - https://github.com/heroiclabs/nakama-dotnet/commit/651930a568473d0995a4c1a65760d1098ee651e8

**Comment by novabyte** [comment]
We've got a release planned for the .NET client code and will then bundle it into the next Unity client.

**Comment by novabyte** [comment]
This is resolved in the [v2.5.0](https://github.com/heroiclabs/nakama-unity/releases/v2.5.0) release.
